extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -294,7 +294,13 @@ export class Bind {
    */
   setStateWithExpression(expression, scope) {
     this.setStatePromise_ = this.evaluateExpression_(expression, scope)
-        .then(result => this.setState(result));
+        .then(result => this.setState(result))
+        .then(() => {
+          this.history_.replace({
+            ampBindState: this.state_,
```

Comment:
We should avoid references to specific extensions from common code like `History`. Instead, we can change this to be a generic JSON object and amp-bind can pass in sub-state for its own use. E.g.

```js
this.history_.replace({
  title: ..., 
  data: {'amp-bind': this.state_},
});
```